### PR TITLE
Correct the objectHash when the value is null

### PR DESCRIPTION
**Pull request by adnasa**
the `jsondiffpatch` executes `objectHash` and may pass in `null` as the `obj` argument value
This PR takes `null` into account and returns the objects hash-index if that were the case.

- [x] commit messages are commitizen-friendly
- [x] code is unit tested
- [x] code is integration tested
- [ ] public code is documented
- [ ] code is reviewed by at least one person
- [x] code satisfies linter rules

---

cc @ct-dali @PhilippSpo

### CODE REVIEW

File: lib/sync/utils/diffpatcher.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ import jsondiffpatch from 'jsondiffpatch'
 
 const diffpatcher = jsondiffpatch.create({
   objectHash (obj, index) {
-    return obj.id || obj.name || obj.url || `$$index:${index}`
+    const objIndex = `$$index:${index}`
+    if (obj === null) return objIndex
```
Comment (ct-dali) [inline lib/sync/utils/diffpatcher.js]: no... what if `obj` is not an object? the following code will crash.

**Comment by emmenko** [comment]
This is not acceptable without a test case that reproduces the problem.

File: test/utils/diffpatcher.spec.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          .map((r, i) => objectHash(r, i))
+
+        t.deepEqual(objectHashIndexValues, [
+          'random1',
+          '$$index:1',
+          '$$index:2',
+          '$$index:3',
+          '$$index:4',
+        ], 'expect index value OR hash of the object in the list')
+        t.end()
```
Comment (adnasa) [inline test/utils/diffpatcher.spec.js]: here is the test coverage

cc @emmenko

Comment (emmenko) [inline test/utils/diffpatcher.spec.js]: Sorry, this is not exactly what I meant. A test with product data that reproduces the error, using buildActions main function. See sync tests, or ask @PhilippSpo

**Comment by PhilippSpo** [comment]
I don't understand why the coverage went down...

**Comment by emmenko** [comment]
@adnasa @PhilippSpo what's the status here?

**Comment by PhilippSpo** [comment]
Coverage of untouched files went down

**Comment by emmenko** [comment]
Don't worry about the coverage

**Comment by PhilippSpo** [comment]
The only thing we could add is an integration test then.

**Comment by adnasa** [comment]
I'll try to find the time today to figure _why_ test coverage went down.
Other than that, I think we are good :)
cc @PhilippSpo

**Comment by adnasa** [comment]
I'd like to have this a go
cc @PhilippSpo @emmenko 

I think one of the main reasons that coverage went down is because more code is added and in the case of http and task-queue utils, some utility were not covered.

Trying to cover them would be out of scope of this PR

**Comment by emmenko** [comment]
🚀

**Comment by adnasa** [comment]
🍹

cc @PhilippSpo @emmenko

**Pull request merged**